### PR TITLE
Feature/1626/view update on checker add

### PR DESCRIPTION
**Pull request by garyluu**
For ga4gh/dockstore#1626

Greatly simplify the sidebar expansion panel and tab selection because there were problems with it for some reason and couldn't figure out what was wrong (probably some inconsistent state).

Update the current selected workflow with the response from the checker workflow

### CODE REVIEW

File: src/app/mytools/my-tool/my-tool.component.ts [hunk 1]
```diff
@@ -89,41 +85,16 @@ export class MyToolComponent extends MyEntry implements OnInit {
     this.registerToolService.tool.subscribe(tool => this.registerTool = tool);
   }
 
-  protected updateActiveTab(): void {
```
Comment (denis-yuen) [inline src/app/mytools/my-tool/my-tool.component.ts]: Nice

File: src/app/shared/entry/expand-panel.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+   *
+   * @param {((OrgToolObject | OrgWorkflowObject))} orgEntriesObject  Either the orgToolObject or orgWorkflowObject
+   * @param {number} entryId  ID of the currently selected workflow
+   * @returns {boolean}  true if open, false if not open
+   * @memberof ExpandPanelPipe
+   */
+  transform(orgEntriesObject: (OrgToolObject | OrgWorkflowObject), entryId: number): boolean {
+    const publishedEntries: Array<ExtendedDockstoreTool | ExtendedWorkflow> = orgEntriesObject.published;
+    const unpublishedEntries: Array<ExtendedDockstoreTool | ExtendedWorkflow> = orgEntriesObject.unpublished;
+    if (publishedEntries.find((entry: Entry) => entry.id === entryId)) {
```
Comment (denis-yuen) [inline src/app/shared/entry/expand-panel.pipe.ts]: Is this just 
`return (publishedEntries.find((entry: Entry) => entry.id === entryId) || (unpublishedEntries.find((entry: Entry) => entry.id === entryId);` ?

File: src/app/shared/entry/select-tab.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  transform(orgEntriesObject: (OrgToolObject | OrgWorkflowObject), entryId: number): number {
+    const publishedEntries: Array<ExtendedDockstoreTool | ExtendedWorkflow> = orgEntriesObject.published;
+    const unpublishedEntries: Array<ExtendedDockstoreTool | ExtendedWorkflow> = orgEntriesObject.unpublished;
+    if (publishedEntries.find((entry: Entry) => entry.id === entryId)) {
+      return 0;
+    }
+    if (unpublishedEntries.find((entry: Entry) => entry.id === entryId)) {
+      return 1;
+    }
+    if (publishedEntries.length > 0) {
```
Comment (denis-yuen) [inline src/app/shared/entry/select-tab.pipe.ts]: `return publishedEntries.length == 0`?

Comment (garyluu) [reply src/app/shared/entry/select-tab.pipe.ts]: needs a number

**Pull request merged**